File: 07-Set-and-Map/Test2/src/LinkedListSet.java
public class LinkedListSet<K extends Comparable<K>> implements Set<K> {

    private LinkedList<K,Object> data;

    public LinkedListSet(){
        data = new LinkedList<>();
    }

    @Override
    public int getSize(){
        return data.getSize();
    }

    @Override
    public boolean isEmpty(){
        return data.isEmpty();
    }

    @Override
    public void add(K key){
        data.add(key,null);
    }

    @Override
    public boolean contains(K key){
        return data.contains(key);
    }

    @Override
    public void remove(K key){
        data.remove(key);
    }

}

File: go/07-Set-and-Map/02-LinkedListSet/linkedlistset/linkedlistset.go
package linkedlistset

import "github.com/dingsw2019/play-with-data-struct/go/07-Set-and-Map/02-LinkedListSet/linkedlist"

type LinkedListSet struct {
	link *linkedlist.LinkedList
}

func New() *LinkedListSet {
	return &LinkedListSet{
		link: linkedlist.New(),
	}
}

func (s *LinkedListSet) Add(e interface{}) {
	if !s.link.Contains(e) {
		s.link.AddFirst(e)
	}
}

func (s *LinkedListSet) Remove(e interface{}) {
	s.link.RemoveElement(e)
}

func (s *LinkedListSet) Contains(e interface{}) bool {
	return s.link.Contains(e)
}

func (s *LinkedListSet) GetSize() int {
	return s.link.GetSize()
}

func (s *LinkedListSet) IsEmpty() bool {
	return s.link.IsEmpty()
}
